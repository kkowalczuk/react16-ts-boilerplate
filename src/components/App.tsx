import * as React from "react"

interface AppProps {
  color: "red" | "blue"
}

export default class App extends React.PureComponent<AppProps> {
  render() {
    return (
      <h1 style={{ backgroundColor: this.props.color }}>
        Welcome!
      </h1>
    )
  }
}
