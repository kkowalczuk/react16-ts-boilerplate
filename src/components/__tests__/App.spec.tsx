import { shallow } from "enzyme"
import "jest"
import * as React from "react"

import App from "../App"

describe("App component", () => {
  it("renders correctly with red background", () => {
    const component = shallow(<App color="red" />)
    expect(component).toMatchSnapshot()
  })

  it("renders correctly with blue background", () => {
    const component = shallow(<App color="blue" />)
    expect(component).toMatchSnapshot()
  })
})
