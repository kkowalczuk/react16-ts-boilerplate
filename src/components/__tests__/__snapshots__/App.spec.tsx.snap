// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly with blue background 1`] = `
<h1
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  Welcome!
</h1>
`;

exports[`App component renders correctly with red background 1`] = `
<h1
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  Welcome!
</h1>
`;
